aten/src/ATen/native/sparse/cuda/SparseBlasImpl.cpp

```diff
@@ -651,6 +651,46 @@ void spmm(
 #endif // !AT_USE_CUSPARSE_GENERIC_API()
 }
 
+void spmm_wrapper(
+    const at::sparse_csr::SparseCsrTensor& mat1,
+    const Tensor& mat2,
+    const Scalar& beta,
+    const Scalar& alpha,
+    const Tensor& result) {
+
+// Cusparse CSR SPMM, starting from cuda 11.6 update 1 (cusparse 11.7.2), has some functionality issue when sparse matrix A is being broadcasted,
+//   by setting both strides to 0 in cusparseCsrSetStridedBatch call.
+// This will be fixed in a future cuda release.
+// We are using a temporary workaround here to do the batched SPMM on every single matrix separately.
+#if CUSPARSE_VERSION < 11702 || CUSPARSE_VERSION >= 11704
+  return spmm(mat1, mat2, beta, alpha, result);
+#else
+  if (mat1.dim() == 2) {
```

Comment:
As we found out the bug is present only for complex inputs. Let's use the workaround only for complex inputs then. I expect a for-loop solution to be slower and no need to slow down the code path for reals if it works fine. A comment above should include info that the bug is present only for complex inputs.
```suggestion
  if (mat1.dim() == 2 || mat1.is_floating_point()) {
```